src/main/java/gov/nasa/pds/api/engineering/controllers/MyProductsApiBareController.java

```diff
@@ -210,7 +211,10 @@ public String getLatestLidVidFromLid(String lid) throws IOException {
               log.error("Couldn't serialize response for content type " + accept, e);
               return new ResponseEntity<Products>(HttpStatus.INTERNAL_SERVER_ERROR);
           }
-            
+        	catch (ParseCancellationException pce) {
+        		log.error("Could not parse the query string: " + q);
+        		return new ResponseEntity<Products>(HttpStatus.BAD_REQUEST);
```

Comment:
That should be easy to add the error message here. You don't need to worry about that as part of this ticket (I created another for that).

Reply:
fixed

Reply:
Thanks @al-niessner 

I was more thinking of something like:
```

  }
        	catch (ParseCancellationException pce) {
        		log.error("Could not parse the query string: " + q);
        		ErrorMessage error = new ErrorMessage();
        		String contextPath = this.request.getContextPath();
        		error.setUrl(this.request.getRequestURL().toString());
        		error.setResource("Products");
        		error.setQuery(this.request.getQueryString());
        		error.setMessage("unable to parse q parameter " + pce.getMessage());
        		return new ResponseEntity<ErrorMessage>(error, HttpStatus.BAD_REQUEST);

        	}            
        }
```

But that does not work yet since the method is expecting ResponseEntity<Products> as a return type. We'll do this work as part of ticket https://github.com/NASA-PDS/registry-api-service/issues/26

Can you remove the update you made here since it does not work either (missing ; plus some other issues I believe).

Thanks

Reply:
not fixed.